### PR TITLE
Renamed baddmm -> addbmm; added baddbmm.

### PR DESCRIPTION
**Pull request by clementfarabet**
- addbmm is a batch MM + reduce add
- baddbmm is a batch MM + batch add

### CODE REVIEW

**Comment by clementfarabet** [comment]
Anyone has any objections to the naming? The idea is to have the "b" prepended to the part of the operator that's batched.

File: lib/TH/generic/THTensorMath.c [hunk 1]
```diff
@@ -709,6 +709,46 @@ void THTensor_(baddmm)(THTensor *result, real beta, THTensor *t, real alpha, THT
   THTensor_(free)(matrix2);
 }
 
+void THTensor_(baddbmm)(THTensor *result, real beta, THTensor *t, real alpha, THTensor *batch1, THTensor *batch2)
```
Comment (dominikgrewe) [inline lib/TH/generic/THTensorMath.c]: We could implement torch.bmm in terms of baddbmm, couldn't we? We just need to make sure that torch.bmm resizes the result tensor before passing it to badbmm.

**Comment by clementfarabet** [comment]
@dominikgrewe that's right – we could simplify a bit like that.

**Comment by soumith** [comment]
naming looks good to me

**Comment by clementfarabet** [comment]
Cool, merging.

**Pull request merged**